### ISSUE TITLE
Broken Scene Broadcasts caused by double pooled writer when network transform is above network object in component order

### ISSUE THREAD

**Issue by tmcarey** [FirstGearGames/FishNet#786]
# Important
If General, Description, and Replication are not completed the issue will be closed immediately.

**General**
Unity version: 
Fish-Networking version: V4.4.7R, Stable
Discord link:
** Above, post a link from our Discord where you troubleshot the issue (on Discord click the three dots next to the message, then Copy Message Link). Issue may be closed out if this is not included.
https://discord.com/channels/424284635074134018/1290684732178432071

**Description**

On network scene objects with network transforms, when the scene changes they get destroyed in order. If the network transform is above network object, it calls network transform destroy first, which stores the network transforms writer but does not null the reference. 

Later, the network object's destroy is called, which calls OnStopNetwork on network transform, which calls ResetState, which calls store again (this time nulling the reference). 

So it gets stored twice leaving a ticking time bomb for the WriteBroadcast function, where it retrieves an additional writer to write the message. Because of the duplicate writers in the pool there is a small chance this message writer is the same writer as the overall broadcast writer, causing the broadcat to be overwritten by the message and corrupting the broadcast.

Anything else that uses two writers like this would be also vulnerable to this issue.

Simple workaround is obviously just to make sure network object is above network transform for all network objects.

**Replication**

Attached, simple project which loads different scenes over and over with network objects that have the network transform above the network object in the component order. Scene change broadcast fails after about 3 loads in most cases.
[FishBugReport.zip](https://github.com/user-attachments/files/17230635/FishBugReport.zip)

**Expected behavior**

Scene broadcast should not fail when network transforms are above network objects.

There should probably be a general backstop against double writers, as these bugs are very hard to track down (as they can be caused by arbitrary code paths) and can be caught with a simple check in the store function. Obviously this has performance costs so this check should probably be behind some debug flags.

**Screenshots**
![image](https://github.com/user-attachments/assets/065e2061-27be-41e8-9fa8-0b45f617f459)

![image](https://github.com/user-attachments/assets/96d1fda6-c0f9-4b59-8a2e-382a4ad77408)

**Comment by FirstGearGames** [comment]
What are the steps to reproduce? I have the package installed but it seems you left out the steps.

**Comment by FirstGearGames** [comment]
Steps:
1) Add all 4 scenes to build settings.
2) Open Start scene, press Play.
3) See error after a few seconds of scene changes.

Notes: Only one networked object with a NetworkTransform needs to exist per scene.

Resolution. Replace these methods in NetworkTransform.

```cs
        private void ResetState()
        {
            _teleport = false;
            ChangeTickSubscription(false);
            /* Reset server and client side since this is called from
             * OnStopNetwork. */

            ObjectCaches<PooledWriter>.StoreAndDefault(ref _authoritativeClientData.Writer);

            WriterPool.StoreAndDefault(ref _toClientChangedWriter);

            ObjectCaches<bool>.StoreAndDefault(ref _authoritativeClientData.HasData);
            ObjectCaches<ChangedDelta>.StoreAndDefault(ref _serverChangedSinceReliable);

            ResettableObjectCaches<TransformData>.StoreAndDefault(ref _lastReceivedClientTransformData);
            ResettableObjectCaches<TransformData>.StoreAndDefault(ref _lastReceivedServerTransformData);
            //Goaldatas. Would only exist if client or clientHost.
            while (_goalDataQueue.Count > 0)
                ResettableObjectCaches<GoalData>.Store(_goalDataQueue.Dequeue());

            ResettableObjectCaches<TransformData>.StoreAndDefault(ref _lastSentTransformData);
            ResettableObjectCaches<GoalData>.StoreAndDefault(ref _currentGoalData);
        }

        /// <summary>
        /// Deinitializes this component for OnDestroy.
        /// </summary>
        private void ResetState_OnDestroy()
        {
            ResettableObjectCaches<TransformData>.StoreAndDefault(ref _lastSentTransformData);
            WriterPool.StoreAndDefault(ref _toClientChangedWriter);
        }
```

**Comment by FirstGearGames** [comment]
Resolved in 4.4.8
